addons/product/models/product_pricelist_item.py

```diff
@@ -177,22 +177,42 @@ def _compute_name_and_price(self):
                     item.env, item.fixed_price, monetary=True, dp="Product Price", currency_obj=item.currency_id)
             elif item.compute_price == 'percentage':
                 percentage = self._get_integer(item.percent_price)
-                item.price = _("%s %% discount", percentage)
-            else:
-                discount_type, percentage = self._get_displayed_discount(item)
-                if not (surcharge := item.price_surcharge):
+                if item.base_pricelist_id:
                     item.price = _(
-                        "%(percentage)s %% %(discount_type)s",
+                        "%(percentage)s %% discount on %(pricelist)s",
                         percentage=percentage,
-                        discount_type=discount_type,
+                        pricelist=item.base_pricelist_id.display_name
                     )
                 else:
-                    item.price = _(
-                        "%(percentage)s %% %(discount_type)s and %(price)s surcharge",
-                        percentage=percentage,
-                        price=surcharge,
-                        discount_type=discount_type,
+                    item.price = _("%(percentage)s %% discount on sales price", percentage=percentage)
+            else:
+                base_str = ""
+                if item.base == 'list_price':
+                    base_str = _("on sales price")
```

Comment:
You can add translation only at the end when you make whole string from part instead add translation to parts this will make translator job easier